src/test/unit_tests/jest_tests/test_cli_upgrade.test.js

```diff
@@ -282,23 +282,21 @@ describe('noobaa cli - upgrade', () => {
         const res = await exec_manage_cli(TYPES.UPGRADE, UPGRADE_ACTIONS.START, { config_root, expected_version: pkg.version, expected_hosts: `${hostname},bla1,bla2` }, true);
         const parsed_res = JSON.parse(res.stdout);
         expect(parsed_res.error.message).toBe(ManageCLIError.UpgradeFailed.message);
-        expect(parsed_res.error.cause).toContain('config dir upgrade can not be started - expected_hosts missing one or more hosts specified in system.json  hosts_data=');
+        expect(parsed_res.error.cause).toContain('config dir upgrade can not be started - expected_hosts contains one or more hosts that are not specified in system.json hosts_data=');
     });
 
-    it('upgrade start - should fail on missing system.json hosts in expected_hosts', async () => {
+    it('upgrade start - should succeed although system.json contains extra hosts than specified in expected_hosts', async () => {
         await fs_utils.replace_file(config_fs.system_json_path, JSON.stringify(invalid_hostname_system_json));
         const res = await exec_manage_cli(TYPES.UPGRADE, UPGRADE_ACTIONS.START, { config_root, expected_version: pkg.version, expected_hosts: `${hostname}` }, true);
-        const parsed_res = JSON.parse(res.stdout);
-        expect(parsed_res.error.message).toBe(ManageCLIError.UpgradeFailed.message);
-        expect(parsed_res.error.cause).toContain('config dir upgrade can not be started - system.json missing one or more hosts info specified in expected_hosts hosts_data');
+        const parsed_res = JSON.parse(res);
+        expect(parsed_res.response.code).toBe(ManageCLIResponse.UpgradeSuccessful.code);
     });
 
-    it('upgrade start - should fail on missing system.json hosts in expected_hosts1', async () => {
+    it('upgrade start - should fail on missing system.json hosts in expected_hosts with comma', async () => {
```

Comment:
I think it should be:
```suggestion
    it('upgrade start - should succeed although on missing system.json hosts in expected_hosts with comma', async () => {
```

I would add a comment near "expected_hosts: `${hostname},`" that we did it intentionally (we know how to parse it... BTW, did we document it somewhere?).